pkg/hubble/relay/observer/server.go

```diff
@@ -78,35 +78,24 @@ func (s *Server) GetFlows(req *observerpb.GetFlowsRequest, stream observerpb.Obs
 
 	g, gctx := errgroup.WithContext(ctx)
 	flows := make(chan *observerpb.GetFlowsResponse, qlen)
-	var connectedNodes, unavailableNodes []string
 
-	for _, p := range peers {
-		if !isAvailable(p.Conn) {
-			s.opts.log.WithField("address", p.Address).Infof(
-				"No connection to peer %s, skipping", p.Name,
-			)
-			unavailableNodes = append(unavailableNodes, p.Name)
-			continue
-		}
-		connectedNodes = append(connectedNodes, p.Name)
-		p := p
-		g.Go(func() error {
-			// retrieveFlowsFromPeer returns blocks until the peer finishes
-			// the request by closing the connection, an error occurs,
-			// or gctx expires.
-			err := retrieveFlowsFromPeer(gctx, s.opts.ocb.observerClient(&p), req, flows)
-			if err != nil {
-				s.opts.log.WithFields(logrus.Fields{
-					"error": err,
-					"peer":  p,
-				}).Warning("Failed to retrieve flows from peer")
+	fc := newFlowCollector(g, req, s.opts)
+	connectedNodes, unavailableNodes := fc.collect(gctx, peers, flows)
+
+	if req.GetFollow() {
+		go func() {
+			updateTimer, updateTimerDone := inctimer.New()
+			defer updateTimerDone()
+			for {
 				select {
-				case flows <- nodeStatusError(err, p.Name):
+				case <-updateTimer.After(s.opts.peerUpdateInterval):
+					peers := s.peers.List()
+					_, _ = fc.collect(ctx, peers, flows)
```

Comment:
Any reason to use `ctx` here instead of `gctx`?
```suggestion
					_, _ = fc.collect(gctx, peers, flows)
```

Reply:
Nope that's just an error. Nice catch!